### PR TITLE
Create/ Edit project

### PR DESCRIPTION
**Pull request by vidyanambiar**
- Form for creating and editing AWX projects
- Minor update to the framework to allow numeric fields to have a minimum value
- Hook `useGetCredentialTypeIDs` to get credential type IDs (required for looking up credentials of a specific type)
- Updates to `PageFormCredentialSelect.tsx` to look up credentials of a specific type, and to support single selection 
- E2E, component tests

image

### CODE REVIEW

**Comment by AlexSCorey** [comment]
On edit in AWX, when the user changes the project type, the sub form resets with empty values.  If the user then returns the project type to the original type, the original values render.  We should probably duplicate this behavior

**Comment by AlexSCorey** [comment]
Found a bug when editing a project from something with scm credential to red hat insights with a redhat insights credential

credentialBug

**Comment by AlexSCorey** [comment]
Projects of type `git` should not have the `cache timeout` option.  This is when editing a project
Screen Shot 2023-05-22 at 10 21 16 AM

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
This is a great start.  Found some bugs and some commented out code to be removed.  Let me know when its ready for another look.

File: frontend/awx/resources/projects/ProjectSubForms/ManualSubForm.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  project_local_paths?: string[];
+}
+
+export function ManualSubForm() {
+  const { t } = useTranslation();
+  const { data: config } = useSWR<IConfigData>(`/api/v2/config/`, (url: string) =>
+    fetch(url).then((r) => r.json())
+  );
+  const brand: string = process.env.BRAND ?? 'AWX';
+  const product: string = process.env.PRODUCT ?? t('Ansible');
```
[the remark this answers is not in the document]
Comment (vidyanambiar) [reply frontend/awx/resources/projects/ProjectSubForms/ManualSubForm.tsx:20]: We have several instances of this in the code. I agree that having a hook would be useful here. It's better addressed separately from this PR I think. I will get an issue opened for it.

**Comment by vidyanambiar** [comment]
@AlexSCorey Thanks for the review! I think I've got all your comments addressed. The bugs you saw seemed related to an earlier comment about resetting the SCM form fields correctly when the source control type is changed.
Could you re-review please?

**Review by AlexSCorey — APPROVED** [review]
I've pulled this down and all of my concerns have been addressed.  Well done!

**Comment by vidyanambiar** [comment]
This was merged in https://github.com/ansible/ansible-ui/pull/581/commits

**Pull request closed without merging**